dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/utils/FileResourceUtils.java

```diff
@@ -207,10 +212,13 @@ public FileResource saveFileResource(String uid, MultipartFile file, FileResourc
     File tmpFile = toTempFile(file);
 
     if (uid != null && fileResourceService.fileResourceExists(uid)) {
-      throw new WebMessageException(
-          conflict(ErrorCode.E1119, FileResource.class.getSimpleName(), uid));
+      throw new ConflictException(ErrorCode.E1119, FileResource.class.getSimpleName(), uid);
+    }
+    if (!jobRunner.isScheduling()) {
+      fileResourceService.syncSaveFileResource(fileResource, Files.readAllBytes(tmpFile.toPath()));
+    } else {
+      fileResourceService.asyncSaveFileResource(fileResource, tmpFile);
```

Comment:
FYI: This is a general fix to make sure file content is available in integration tests, the asynchronous save was not happening in test setup but at the same time I did not want to change this for prod as it might affect many callers.